etc/shell.nix

```diff
@@ -33,6 +38,63 @@ clangStdenv.mkDerivation rec {
     # slow as it behaves as if -j1 was passed.
     # See https://github.com/servo/mozjs/issues/375
     pinnedNixpkgs.gnumake
+
+    # crown needs to be in our Cargo workspace so we can test it with `mach test`. This means its
+    # dependency tree is listed in the main Cargo.lock, making it awkward to build with Nix because
+    # all of Servo’s dependencies get pulled into the Nix store too, wasting over 1GB of disk space.
+    # Filtering the lockfile to only the parts needed by crown saves space and builds faster.
+    (let
+      vendorTarball = rustPlatform.fetchCargoTarball {
+        src = ../support/filterlock;
+        hash = "sha256-/kJNDtmv2uI7Qlmpi3DMWSw88rzEJSbroO0/QrgQrSc=";
+      };
+      vendorConfig = builtins.toFile "toml" ''
+        [source.crates-io]
+        replace-with = "vendor"
+        [source.vendor]
+        directory = "vendor"
+      '';
+
+      # Build and run filterlock over the main Cargo.lock.
+      filteredLockFile = (clangStdenv.mkDerivation {
+        name = "lock";
+        buildInputs = [ cargo ];
+        src = ../support/filterlock;
+        buildPhase = ''
+          tar xzf ${vendorTarball}
+          mv cargo-deps-vendor.tar.gz vendor
+          mkdir .cargo
+          cp -- ${vendorConfig} .cargo/config.toml
+          > $out cargo run --offline -- ${../Cargo.lock} crown
+        '';
+      });
+    in (rustPlatform.buildRustPackage rec {
+      name = "crown";
+      src = ../support/crown;
+      stdenv = nixpkgs_rust_1_74.clangStdenv;
+      buildInputs = [ nixpkgs_rust_1_74.llvmPackages.libllvm ];
+      doCheck = false;
+      cargoLock = {
+        lockFileContents = builtins.readFile filteredLockFile;
+
+        # Needed when filteredLockFile = ../Cargo.lock.
+        # allowBuiltinFetchGit = true;
```

Comment:
I am not sure I understand in which scenario we will want this to be uncommented. Maybe the comment above can be expanded a bit?

Reply:
done in dca7c89526e3c66b09e4a72b0fddb0990706214b